templates/vm.xml.j2

```diff
@@ -1,38 +1,45 @@
 <domain type='{{ libvirt_vm_engine }}'>
-  <name>{{ libvirt_vm_name }}</name>
-  <memory>{{ libvirt_vm_memory_mb | int * 1024 }}</memory>
-  <vcpu>{{ libvirt_vm_vcpus }}</vcpu>
+  <name>{{ vm.name }}</name>
+  <memory>{{ vm.memory_mb | int * 1024 }}</memory>
+  <vcpu>{{ vm.vcpus }}</vcpu>
   <clock sync="localtime"/>
+  <on_poweroff>destroy</on_poweroff>
+  <on_reboot>restart</on_reboot>
+  <on_crash>destroy</on_crash>
   <os>
-    <type arch='{{ libvirt_vm_arch }}'{% if libvirt_vm_machine is not none %} machine='{{ libvirt_vm_machine }}'{% endif %}>hvm</type>
+    <type arch='{{ libvirt_vm_arch }}'{% if vm.machine is defined %} machine='{{ vm.machine | default('pc-1.0') }}'{% endif %}>hvm</type>
     <bootmenu enable='no'/>
+    <boot dev='hd'/>
+    <boot dev='network'/>
```

Comment:
I guess this will continue to work for non network-booted VMs?

Reply:
Looks like when i boot a VM with a hard drive and no network interface, it successfully boots from the hard drive even with this configuration.